client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -706,6 +834,16 @@ private TxnSegments convert(CreateTxnResponse response) {
                 });
     }
 
+    @Override
+    public void close() {
+        if (closed.compareAndSet(false, true)) {
+            final ScheduledExecutorService executorService = this.internalExecutor.getAndSet(null);
+            if (executorService != null) {
+                executorService.shutdownNow();
```

Comment:
If this class did not create the thread pool, it cannot be sure that the caller isn't also holding onto it and using it. So it should not close it. This needs to be documented of course.